src/components/carousel/carousel-indicator.ts

```diff
@@ -0,0 +1,56 @@
+import { LitElement, html } from 'lit';
+import { property } from 'lit/decorators.js';
+import { styleMap } from 'lit/directives/style-map.js';
+import { registerComponent } from '../common/definitions/register.js';
+import { styles } from './themes/carousel-indicator.base.css.js';
+
+/**
+ * Used when a custom indicator needs to be passed to the `igc-carousel` component.
+ *
+ * @element igc-carousel-indicator
+ *
+ * @slot - Default slot for projected inactive indicator.
+ * @slot active - Default slot for projected active indicator.
+ *
+ */
+export default class IgcCarouselIndicatorComponent extends LitElement {
+  public static readonly tagName = 'igc-carousel-indicator';
+  public static override styles = styles;
+
+  /* blazorSuppress */
+  public static register() {
+    registerComponent(IgcCarouselIndicatorComponent);
+  }
+
+  /** @private @hidden @internal */
+  @property()
+  public active = false;
```

Comment:
```suggestion
  @property({ attribute: false })
  public active = false;
```
Don't see a point to go through hoops just to hide it from the API docs. Just leave it as such.